firewood/src/hashednode.rs

```diff
@@ -50,15 +53,24 @@ impl<T: ReadLinearStore> From<NodeStore<T>> for HashedNodeStore<T> {
 impl<T: ReadLinearStore> HashedNodeStore<T> {
     pub fn read_node(&self, addr: LinearAddress) -> Result<Arc<Node>, Error> {
         if let Some(modified_node) = self.modified.get(&addr) {
-            Ok(modified_node.0.clone())
+            if let Some((modified_node, _)) = modified_node {
+                Ok(modified_node.clone())
+            } else {
+                Err(Error::new(std::io::ErrorKind::Other, "Node not found"))
```

Comment:
Although I think this code shouldn't be needed at all (see earlier comment) at a minimum this should be the same as when a node is read that has been freed, see `read_node` in nodestore which does this:
```
Err(Error::new(
                ErrorKind::InvalidData,
                "Attempted to read a freed area",
            )),
```